File: Array/easy/maximum_consecutive_ones.cpp
//
// Created by santosh on 12/10/25.
//

#include<bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maximumConsecutiveOnes(vector<int>& nums)
    {
        int maxCount = 0;
        int count = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] == 0)
            {
                maxCount = max(maxCount, count);
                count = 0;
            }
            else
            {
                count++;
                maxCount = max(maxCount, count);
            }
        }
        return maxCount;
    }
};

int main()
{
    // vector<int> nums = {1, 0, 1, 1, 0, 1};
    vector<int> nums= {1, 1, 0, 1, 1, 1};
    Solution s;

    int res = s.maximumConsecutiveOnes(nums);
    cout << res << endl;
}

File: Array/easy/maximum_consecutive_ones.py
# Maximum consecutive ones in a binary array
# Created by santosh

class Solution:
    def maximum_consecutive_ones(self, nums: list[int]) -> int:
        max_count = 0
        count = 0
        for num in nums:
            if num == 0:
                max_count = max(max_count, count)
                count = 0
            else:
                count += 1
                max_count = max(max_count, count)
        return max_count


if __name__ == "__main__":
    nums = [1, 1, 0, 1, 1, 1]
    s = Solution()
    print(s.maximum_consecutive_ones(nums))
